### DISCUSSION TITLE
Оптимизация количества запросов к Lastfm

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#17] [Идея] [answered]
Добрый день!

Я так понимаю, это нормальная ситуация в плане числа запросов? VERSION = '1.3.1'
С 20 декабря работает тригер по обновлению истории с Last Fm.
![image](https://user-images.githubusercontent.com/76212904/103123503-9e9b9580-4695-11eb-87fc-8f4153e2785a.png)

**Comment by Chimildic** [comment, marked as answer]
Квота запросов Apps Script - 20 тысяч в день. Про все ограничения написано [здесь](https://chimildic.github.io/goofy/#/desc?id=Ограничения). Или вопрос о чем-то другом?

**Reply by efremych** [reply]
Значительный рост запросов просто. Которые в основном проходят вхолостую, как понимаю - поиск 30 треков от Last Fm в базе Spotify каждые 15 минут.

Может имеет смысл, например, сделать еще один кэш для Last Fm, где данные будут в формате, который отдает Last FM. И при получении 30 треков от Last Fm сверять их с кэшем. И только если обнаружатся треки, отсутствующие в данном кэше, запускать остальные процедуры по данным трекам (поиск в Spotify, перезапись основного кэшаLast FM)

**Reply by Chimildic** [reply]
Хорошее замечание, посмотрю в эту сторону. UPD: появится в 1.3.3
